src/runtime_src/driver/xclng/xrt/user_gem/shim.cpp

```diff
@@ -331,10 +333,36 @@ int xocl::XOCLShim::pcieBarWrite(unsigned int pf_bar, unsigned long long offset,
  */
 int xocl::XOCLShim::xclLogMsg(xclDeviceHandle handle, xclLogMsgLevel level, const char* format, ...)
 {
-    //This is TODO
-    xrt_core::message::send((xrt_core::message::severity_level)level, format);
+    va_list args1;
+    va_start(args1, format);
+    int len = std::vsnprintf(nullptr, 0, format, args1);
+    va_end(args1);
+    
+    {
+        if (len < 0) {
+            goto handle_errors;
```

Comment:
No goto in C++ code. Please move the code inside handle_errors label here.

Reply:
Done